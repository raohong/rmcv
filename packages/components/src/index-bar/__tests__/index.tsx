import { render } from '@testing-library/react';
import { IndexAnchor, IndexBar } from '..';

it('render correctly', () => {
  const tree = render(
    <IndexBar>
      <IndexAnchor index='A' />
      <p>A1</p>
      <IndexAnchor index='B' />
      <p>B1</p>
    </IndexBar>,
  );

  expect(tree.asFragment()).toMatchSnapshot();
});
