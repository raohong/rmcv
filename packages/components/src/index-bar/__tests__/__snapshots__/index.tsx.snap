// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-tap-highlight-color: transparent;
  -webkit-font-smooth: antialiased;
  position: relative;
  touch-action: none;
  list-style: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,Helvetica Neue,Helvetica,Segoe UI,Arial,Roboto,PingFang SC,miui,Hiragino Sans GB,Microsoft Yahei,sans-serif;
}

.emotion-1 {
  box-sizing: border-box;
}

.emotion-3 {
  box-sizing: border-box;
  padding: 0 16px;
  color: #323233;
  font-weight: 500;
  font-size: 14px;
  line-height: 32px;
  background: transparent;
}

.emotion-7 {
  position: fixed;
  box-sizing: border-box;
  top: 50%;
  right: 0;
  touch-action: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  padding: 0 8px 0 16px;
  z-index: 1;
}

.emotion-8 {
  cursor: pointer;
  box-sizing: border-box;
  color: #323233;
  font-weight: 500;
  line-height: 20px;
  text-align: center;
}

.emotion-9 {
  cursor: pointer;
  box-sizing: border-box;
  color: #1989fa;
  font-weight: 500;
  line-height: 20px;
  text-align: center;
}

<div
    class="RVuiIndexBar-root emotion-0"
  >
    <div>
      <div
        class="RVuiSticky-root emotion-1"
      >
        <div
          class="RVuiSticky-content emotion-1"
        >
          <div
            class="RVuiIndexBar-anchor emotion-3"
          >
            A
          </div>
        </div>
      </div>
    </div>
    <p>
      A1
    </p>
    <div>
      <div
        class="RVuiSticky-root emotion-1"
      >
        <div
          class="RVuiSticky-content emotion-1"
        >
          <div
            class="RVuiIndexBar-anchor emotion-3"
          >
            B
          </div>
        </div>
      </div>
    </div>
    <p>
      B1
    </p>
    <div
      class="RVuiIndexBar-sidebar emotion-7"
    >
      <div
        aria-label="A"
        class="RVuiIndexBar-index emotion-8"
        data-index=""A""
      >
        A
      </div>
      <div
        aria-label="B"
        class="RVuiIndexBar-index emotion-9"
        data-index=""B""
      >
        B
      </div>
      <div
        aria-label="C"
        class="RVuiIndexBar-index emotion-8"
        data-index=""C""
      >
        C
      </div>
      <div
        aria-label="D"
        class="RVuiIndexBar-index emotion-8"
        data-index=""D""
      >
        D
      </div>
      <div
        aria-label="E"
        class="RVuiIndexBar-index emotion-8"
        data-index=""E""
      >
        E
      </div>
      <div
        aria-label="F"
        class="RVuiIndexBar-index emotion-8"
        data-index=""F""
      >
        F
      </div>
      <div
        aria-label="G"
        class="RVuiIndexBar-index emotion-8"
        data-index=""G""
      >
        G
      </div>
      <div
        aria-label="H"
        class="RVuiIndexBar-index emotion-8"
        data-index=""H""
      >
        H
      </div>
      <div
        aria-label="I"
        class="RVuiIndexBar-index emotion-8"
        data-index=""I""
      >
        I
      </div>
      <div
        aria-label="J"
        class="RVuiIndexBar-index emotion-8"
        data-index=""J""
      >
        J
      </div>
      <div
        aria-label="K"
        class="RVuiIndexBar-index emotion-8"
        data-index=""K""
      >
        K
      </div>
      <div
        aria-label="L"
        class="RVuiIndexBar-index emotion-8"
        data-index=""L""
      >
        L
      </div>
      <div
        aria-label="M"
        class="RVuiIndexBar-index emotion-8"
        data-index=""M""
      >
        M
      </div>
      <div
        aria-label="N"
        class="RVuiIndexBar-index emotion-8"
        data-index=""N""
      >
        N
      </div>
      <div
        aria-label="O"
        class="RVuiIndexBar-index emotion-8"
        data-index=""O""
      >
        O
      </div>
      <div
        aria-label="P"
        class="RVuiIndexBar-index emotion-8"
        data-index=""P""
      >
        P
      </div>
      <div
        aria-label="Q"
        class="RVuiIndexBar-index emotion-8"
        data-index=""Q""
      >
        Q
      </div>
      <div
        aria-label="R"
        class="RVuiIndexBar-index emotion-8"
        data-index=""R""
      >
        R
      </div>
      <div
        aria-label="S"
        class="RVuiIndexBar-index emotion-8"
        data-index=""S""
      >
        S
      </div>
      <div
        aria-label="T"
        class="RVuiIndexBar-index emotion-8"
        data-index=""T""
      >
        T
      </div>
      <div
        aria-label="U"
        class="RVuiIndexBar-index emotion-8"
        data-index=""U""
      >
        U
      </div>
      <div
        aria-label="V"
        class="RVuiIndexBar-index emotion-8"
        data-index=""V""
      >
        V
      </div>
      <div
        aria-label="W"
        class="RVuiIndexBar-index emotion-8"
        data-index=""W""
      >
        W
      </div>
      <div
        aria-label="X"
        class="RVuiIndexBar-index emotion-8"
        data-index=""X""
      >
        X
      </div>
      <div
        aria-label="Y"
        class="RVuiIndexBar-index emotion-8"
        data-index=""Y""
      >
        Y
      </div>
      <div
        aria-label="Z"
        class="RVuiIndexBar-index emotion-8"
        data-index=""Z""
      >
        Z
      </div>
    </div>
  </div>
</DocumentFragment>
`;
